python/lsst/ts/ATWhiteLightSource/wlsCSC.py

```diff
@@ -369,29 +411,105 @@ def get_config_pkg():
+                self.last_warning_state["Process Fluid Level Warning"] = False
+
+            if "Switch to Supply Temp as Control Temp Warning" in self.chillerModel.warnings:
+                self.evt_chillerSwitchToSupplyTempWarning.set_put(timestamp=time.time(), warning=True)
+                self.last_warning_state["Switch to Supply Temp as Control Temp Warning"] = True
+            elif "Switch to Supply Temp as Control Temp Warning" in self.last_warning_state and self.last_warning_state["Switch to Supply Temp as Control Temp Warning"]:
+                self.evt_chillerSwitchToSupplyTempWarning.set_put(timestamp=time.time(), warning=False)
+                self.last_warning_state["Switch to Supply Temp as Control Temp Warning"] = False
+
+            if "High Control Temp Warning" in self.chillerModel.warnings:
+                self.evt_chillerHighControlTempWarning.set_put(timestamp=time.time(), warning=True)
+                self.last_warning_state["High Control Temp Warning"] = True
+            elif "High Control Temp Warning" in self.last_warning_state and self.last_warning_state["High Control Temp Warning"]:
+                self.evt_cchillerHighControlTempWarning.set_put(timestamp=time.time(), warning=False)
+                self.last_warning_state["High Control Temp Warning"] = False
+
+            if "Low Control Temp Warning" in self.chillerModel.warnings:
+                self.evt_chillerLowControlTempWarning.set_put(timestamp=time.time(), warning=True)
+                self.last_warning_state["Low Control Temp Warning"] = True
+            elif "Low Control Temp Warning" in self.last_warning_state and self.last_warning_state["Low Control Temp Warning"]:
+                self.evt_chillerLowControlTempWarning.set_put(timestamp=time.time(), warning=False)
+                self.last_warning_state["Low Control Temp Warning"] = False
+
+            if "High Ambient Temp Warning" in self.chillerModel.warnings:
+                self.evt_chillerHighAmbientTempWarning.set_put(timestamp=time.time(), warning=True)
+                self.last_warning_state["High Ambient Temp Warning"] = True
+            elif "High Ambient Temp Warning" in self.last_warning_state and self.last_warning_state["High Ambient Temp Warning"]:
+                self.evt_cchillerHighAmbientTempWarning.set_put(timestamp=time.time(), warning=False)
+                self.last_warning_state["High Ambient Temp Warning"] = False
+
+            if "Low Ambient Temp Warning" in self.chillerModel.warnings:
+                self.evt_chillerLowAmbientTempWarning.set_put(timestamp=time.time(), warning=True)
+                self.last_warning_state["Low Ambient Temp Warning"] = True
+            elif "Low Ambient Temp Warning" in self.last_warning_state and self.last_warning_state["Low Ambient Temp Warning"]:
+                self.evt_chillerLowAmbientTempWarning.set_put(timestamp=time.time(), warning=False)
```

Comment:
I think `timestamp` is no longer a valid topic attribute! Also, the topic if filled with several timestamps by salobj.